test/memoize.js

```diff
@@ -30,4 +30,25 @@ describe('memoize', function() {
         assert.strictEqual(identity('x'), 'x');
         assert.strictEqual(identity('y'), 'y');
     });
+
+    it('can memoize falsy values', function() {
+        var ctr = 0;
+        var f = R.memoize(function(x) { ctr++; return x; });
+        var values = [0, false, null, undefined, '', NaN];
+        checkFunctionIsIdentityOn(values);
+        // check function has been called once per value
+        assert.equal(values.length, ctr);
+        checkFunctionIsIdentityOn(values);
+        // check function has not been called again for any of the values
+        assert.equal(values.length, ctr);
+        function checkFunctionIsIdentityOn(values) {
+            values.forEach(function(v) {
```

Comment:
I'm new to JavaScript. Is Array.forEach() not supported in ES3?

Peter

On 6 February 2015 at 01:20, Michael Hurley <notifications@github.com>
wrote:

> In test/memoize.js
> <https://github.com/ramda/ramda/pull/776#discussion_r24216425>:
>
> > @@ -30,4 +30,25 @@ describe('memoize', function() {
> >          assert.strictEqual(identity('x'), 'x');
> >          assert.strictEqual(identity('y'), 'y');
> >      });
> > +
> > +    it('can memoize falsy values', function() {
> > +        var ctr = 0;
> > +        var f = R.memoize(function(x) { ctr++; return x; });
> > +        var values = [0, false, null, undefined, '', NaN];
> > +        checkFunctionIsIdentityOn(values);
> > +        // check function has been called once per value
> > +        assert.equal(values.length, ctr);
> > +        checkFunctionIsIdentityOn(values);
> > +        // check function has not been called again for any of the values
> > +        assert.equal(values.length, ctr);
> > +        function checkFunctionIsIdentityOn(values) {
> > +            values.forEach(function(v) {
>
> this will fail in ES3
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ramda/ramda/pull/776/files#r24216425>.
>